### ISSUE TITLE
Allow for user-provided GCP bucket encryption key

### ISSUE THREAD

**Issue by ericsmalling** [thelastpickle/cassandra-medusa#773]
When the GCS Bucket is encrypted with customer-managed key (which is what my client does), GCS list will not include the field `md5Hash` in the response.  Therefore, they want to use the `etag` field instead of `md5Hash`.

**Comment by c3-clement** [comment]
Indeed, when the GCS bucket is encrypted with customer-managed key, the List operation from the GCS API won't expose the `md5Hash` field, which cause Medusa code to crash there https://github.com/thelastpickle/cassandra-medusa/blob/master/medusa/storage/google_storage.py#L81 .



Note that this behavior is not documented anywhere, but I opened a ticket to GCP support.
See our discussion below

![support-1](https://github.com/thelastpickle/cassandra-medusa/assets/111791574/7ac7299f-68d9-4241-8195-3f82cbf1382e)

![support-2](https://github.com/thelastpickle/cassandra-medusa/assets/111791574/e6f37acf-d6f7-4715-91fb-2a870f52abe6)

**Comment by c3-clement** [comment]
GCP team also have ticket about this implemented since 2018:
https://issuetracker.google.com/issues/115290740#comment1

**Comment by ericsmalling** [comment]
Returning from a break, will be working on this asap
